apps/emqx/src/emqx_persistent_session_ds.hrl

```diff
@@ -65,7 +65,9 @@
     last_seqno_qos2 = 0 :: emqx_persistent_session_ds:seqno(),
     %% This stream belongs to an unsubscribed topic-filter, and is
     %% marked for deletion:
-    unsubscribed = false :: boolean()
+    unsubscribed = false :: boolean(),
+    %% Reference to the subscription state:
+    sub_state_id :: emqx_persistent_session_ds_subs:subscription_state_id()
```

Comment:
Do you mean why not store the key-value map of states in the subscription record itself?
Separation helps with the following concerns:
- Unsubscribing logic is much simpler: now it's possible to simply delete or update the subscription record, and it won't affect the replay since the SRS records reference subscription states instead of subscriptions.
- Replay (hot path) logic is also simpler: it only needs one lookup to get the subopts.

Reply:
> Do you mean why not store the key-value map of states in the subscription record itself?

No, I was thinking about SRS record: what's the goal of introducing subscription state table and referencing it as opposed to just storing Subopts + UpgradeQoS there?

Reply:
SRS record should be as lean as possible. While it's possible to pack the existing subscription state efficiently now, relying on this is not future proof.